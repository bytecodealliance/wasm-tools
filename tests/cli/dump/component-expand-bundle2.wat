;; RUN: dump %

(component
  (component $c
    (core module (export "e"))
  )
  (component $c2 (import "i" (component (import "i" (core module)))))
  (instance $C (instantiate $c))
  (alias export $C "e" (core module $m))
  (instance (instantiate $c2 (with "i" (instance
    (export "e" (core module $m))
  ))))
)
