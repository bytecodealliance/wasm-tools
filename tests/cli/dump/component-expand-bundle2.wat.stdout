  0x0 | 00 61 73 6d | version 12 (Component)
      | 0c 00 01 00
  0x8 | 04 31       | [component 0] inline size
    0xa | 00 61 73 6d | version 12 (Component)
        | 0c 00 01 00
   0x12 | 01 08       | [core module 0] inline size
     0x14 | 00 61 73 6d | version 1 (Module)
          | 01 00 00 00
   0x1c | 0b 08       | component export section
   0x1e | 01          | 1 count
   0x1f | 01 65 00 00 | export ComponentExport { name: "e", url: "", kind: Module, index: 0, ty: None }
        | 11 00 00   
   0x26 | 00 13       | custom section
   0x28 | 0e 63 6f 6d | name: "component-name"
        | 70 6f 6e 65
        | 6e 74 2d 6e
        | 61 6d 65   
   0x37 | 00 02       | component name
   0x39 | 01 63       | "c"
 0x3b | 04 35       | [component 1] inline size
   0x3d | 00 61 73 6d | version 12 (Component)
        | 0c 00 01 00
   0x45 | 07 0d       | component type section
   0x47 | 01          | 1 count
   0x48 | 41 02 00 50 | [type 0] Component([CoreType(Module([])), Import(ComponentImport { name: "i", url: "", ty: Module(0) })])
        | 00 03 01 69
        | 00 00 11 00
   0x54 | 0a 06       | component import section
   0x56 | 01          | 1 count
   0x57 | 01 69 00 04 | [component 0] ComponentImport { name: "i", url: "", ty: Component(0) }
        | 00         
   0x5c | 00 14       | custom section
   0x5e | 0e 63 6f 6d | name: "component-name"
        | 70 6f 6e 65
        | 6e 74 2d 6e
        | 61 6d 65   
   0x6d | 00 03       | component name
   0x6f | 02 63 32    | "c2"
 0x72 | 05 04       | component instance section
 0x74 | 01          | 1 count
 0x75 | 00 00 00    | [instance 0] Instantiate { component_index: 0, args: [] }
 0x78 | 06 07       | component alias section
 0x7a | 01          | 1 count
 0x7b | 00 11 00 00 | alias [module 0] InstanceExport { kind: Module, instance_index: 0, name: "e" }
      | 01 65      
 0x81 | 05 0f       | component instance section
 0x83 | 02          | 2 count
 0x84 | 01 01 01 65 | [instance 1] FromExports([ComponentExport { name: "e", url: "", kind: Module, index: 0, ty: None }])
      | 00 11 00   
 0x8b | 00 01 01 01 | [instance 2] Instantiate { component_index: 1, args: [ComponentInstantiationArg { name: "i", kind: Instance, index: 1 }] }
      | 69 05 01   
 0x92 | 00 29       | custom section
 0x94 | 0e 63 6f 6d | name: "component-name"
      | 70 6f 6e 65
      | 6e 74 2d 6e
      | 61 6d 65   
 0xa3 | 01 06 00 11 | core module section
 0xa7 | 01          | 1 count
 0xa8 | 00 01 6d    | Naming { index: 0, name: "m" }
 0xab | 01 09 04    | component section
 0xae | 02          | 2 count
 0xaf | 00 01 63    | Naming { index: 0, name: "c" }
 0xb2 | 01 02 63 32 | Naming { index: 1, name: "c2" }
 0xb6 | 01 05 05    | instance section
 0xb9 | 01          | 1 count
 0xba | 00 01 43    | Naming { index: 0, name: "C" }
